### DISCUSSION TITLE
Is any possible the report to be generate only for fail tests not when the test cases are passed?

### DISCUSSION THREAD

**Discussion by CosteaIonut89** [pytest-dev/pytest-html#528] [Q&A]
I am using this command `pytest -v -s --html=report.html --self-contained-html --capture=tee-sys  -rx --verbose --maxfail=6  testCases/test_signup_0.py
`
and I want the report to be generated only when the test cases fail not when the test cases pass. I tried to edit in the conftest but no results in that way.

Thank you in advance!

**Comment by BeyondEvil** [comment]
Just to clarify:

Do you want the entire report to only be generated if there are failures?

Or you do you only want failures _in_ the report?

**Reply by CosteaIonut89** [reply]
Sorry for the misunderstandings.
I want the report to only be generated if one test fails like example I got:
Test1 - Pass
Test2 - Pass
The report should no be generated but if the 
Test1 -Fail
Test2 -Pass
The report should be generated

**Reply by BeyondEvil** [reply]
Interesting.

I'm sure we can accommodate this. I'll ponder the implementation and add it as a feature for v4.x.

**Reply by BeyondEvil** [reply]
What about:

- reruns
- xfail
- xpass
- skipped
?

**Reply by CosteaIonut89** [reply]
I am sure these can be custom too, to try to set if want to ignore 1 of this status.
